rpc/errors.go

```diff
@@ -46,205 +46,151 @@ func Err(code int, data any) *RPCError {
 // - rpcErrors: variadic list of *RPCError objects to be checked
 // Returns:
 // - error: the original error
-func tryUnwrapToRPCErr(err error, rpcErrors ...*RPCError) error {
-	var nodeErr *RPCError
-	if json.Unmarshal([]byte(err.Error()), nodeErr) != nil {
-		return err
+func tryUnwrapToRPCErr(err error, rpcErrors ...*RPCError) *RPCError {
+	errBytes, errIn := json.Marshal(err)
```

Comment:
`json.Marshal()` is not equivalent to `[]byte(err.Error())`, was this intentional?

Reply:
Yeah it's intentional (but it could use a comment). The old code was incorrect because CallContext would return a `ethrpc.DataError`, and `Error()` would just return the main part of the message (e.g. "Contract error" but no context). This code seems to try and remap from `DataError` to a RPCError. We could do this a bit more explicitly by casting to `DataError` and remapping the types

Reply:
The reason I took the marshalling approach was so that it would work with any type with a {Code, Message, Data} structure (DataError, RPCError, any other error that has a similar structure), whereas explicit casting seemed a bit more constrained.